src-vue/src/plugins/store.ts

```diff
@@ -0,0 +1,193 @@
+            release_canal: ReleaseCanal.RELEASE,
+
+            northstar_is_running: false,
+            origin_is_running: false
+        }
+    },
+    mutations: {
+        toggleDeveloperMode(state) {
+            state.developer_mode = !state.developer_mode;
+
+            // Reset tab when closing dev mode.
+            if (!state.developer_mode) {
+                store.commit('updateCurrentTab', Tabs.PLAY);
+            }
+        },
+        initialize(state) {
+            _initializeApp(state);
+            // _checkForFlightCoreUpdates(state);
+            _initializeListeners(state);
+        },
+        updateCurrentTab(state: any, newTab: Tabs) {
+            state.current_tab = newTab;
+        },
+        async launchGame(state: any) {
+            // TODO update installation if release track was switched
+
+            // Install northstar if it wasn't detected.
+            if (state.northstar_state === NorthstarState.INSTALL) {
+                let install_northstar_result = invoke("install_northstar_caller", { gamePath: state.game_path, northstarPackageName: ReleaseCanal.RELEASE });
+                state.northstar_state = NorthstarState.INSTALLING;
+
+                await install_northstar_result.then((message) => {
+                    console.log(message);
+                })
+                    .catch((error) => {
+                        console.error(error);
+                        alert(error);
+                    });
+
+                _get_northstar_version_number(state);
+                return;
+            }
+
+            // Update northstar if it is outdated.
+            if (state.northstar_state === NorthstarState.MUST_UPDATE) {
+                // Updating is the same as installing, simply overwrites the existing files
+                let install_northstar_result = invoke("install_northstar_caller", { gamePath: state.game_path, northstarPackageName: ReleaseCanal.RELEASE });
+                state.northstar_state = NorthstarState.UPDATING;
+
+                await install_northstar_result.then((message) => {
+                    console.log(message);
+                })
+                    .catch((error) => {
+                        console.error(error);
+                        alert(error);
+                    });
+
+                _get_northstar_version_number(state);
+                return;
+            }
+
+            // Game is ready to play
+            if (state.northstar_state === NorthstarState.READY_TO_PLAY) {
+                // Show an error message if Origin is not running.
+                if (!state.origin_is_running) {
+                    ElNotification({
+                        title: 'Origin is not running',
+                        message: "Northstar cannot launch while you're not authenticated with Origin.",
+                        type: 'warning',
+                        position: 'bottom-right'
+                    });
```

Comment:
This is done on back-end anyway so can be removed.

Just add the custom alert notification into the catch of the backend result so it shows as that instead of the JS default alert popup.

Reply:
Done with 98d0e35e5a6e9966406e968e2d247e5e9b6f7768